test/perf/test_performance.py

```diff
@@ -0,0 +1,86 @@
+
+from conftest import get_all_perf_test_configurations
+
+all_perf_test_configurations = get_all_perf_test_configurations()
+all_perf_test_configurations_descriptions = \
+    [config.__repr__().replace(' ', '') for config in all_perf_test_configurations]
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.mark.parametrize("perf_test_configuration", all_perf_test_configurations,
+                         ids=all_perf_test_configurations_descriptions)
+def test_perf(perf_test_configuration):
+    print("Downloading dataset")
+    dataset = datasets[perf_test_configuration.dataset]()
+    print("Done downloading dataset")
+
+    if perf_test_configuration.dataset == "adult_uci":
+        # sensitive feature is 8th column (sex)
+        sensitive_features_train = dataset.X_train[:, 7]
+        sensitive_features_test = dataset.X_test[:, 7]
+    elif perf_test_configuration.dataset == "diabetes_sklearn":
+        # sensitive feature is 2nd column (sex)
+        # features have been scaled, but sensitive feature needs to be str or int
+        sensitive_features_train = dataset.X_train[:, 1].astype(str)
+        sensitive_features_test = dataset.X_test[:, 1].astype(str)
+        # labels can't be floats as of now
+        dataset.y_train = dataset.y_train.astype(int)
+        dataset.y_test = dataset.y_test.astype(int)
+    elif perf_test_configuration.dataset == "compas":
+        # sensitive feature is either race or sex
+        # TODO add another case where we use sex as well, or both (?)
+        sensitive_features_train = dataset.race_train
+        sensitive_features_test = dataset.race_test
+    else:
+        raise ValueError("Sensitive features unknown for dataset {}"
+                         .format(perf_test_configuration.dataset))
+
+    print("Fitting estimator")
+    estimator = models[perf_test_configuration.predictor]()
+    unconstrained_predictor = models[perf_test_configuration.predictor]()
+    unconstrained_predictor.fit(dataset.X_train, dataset.y_train)
+    print("Done fitting estimator")
+
+    start_time = time()
+    if perf_test_configuration.mitigator == ThresholdOptimizer.__name__:
+        mitigator = ThresholdOptimizer(unconstrained_model=unconstrained_predictor,
+                                       parity_criteria=DEMOGRAPHIC_PARITY,
+                                       random_state=1)
+    elif perf_test_configuration.mitigator == ExponentiatedGradient.__name__:
+        mitigator = ExponentiatedGradient(estimator=estimator,
+                                          constraints=DemographicParity())
+    elif perf_test_configuration.mitigator == GridSearch.__name__:
+        mitigator = GridSearch(estimator=estimator,
+                               constraints=DemographicParity())
+    else:
+        raise Exception("Unknown mitigation technique.")
+
+    print("Fitting mitigator")
+
+    mitigator.fit(dataset.X_train, dataset.y_train, sensitive_features=sensitive_features_train)
```

Comment:
As a follow up, it would be good if our fairlearn estimators could also keep track of how many times they called `fit` on the supplied estimator. Even the actual time spent there as well. For performance testing of fairlearn, we just want to make sure that we're not doing anything stupid; most of the time should be spend in the raw estimator.

Reply:
Yes! I'll count all your great suggestions as follow-up tasks/feature requests, because this needs to go in quickly so that I can use the same for notebook examples.